tt_metal/impl/program/program.cpp

```diff
@@ -78,38 +78,205 @@ size_t KernelCompileHash(const std::shared_ptr<Kernel> kernel, JitBuildOptions &
 }  // namespace
 namespace detail {
 
+class Program_ {
```

Comment:
I assume the trailing "_" is an "internal" designation?  I don't think we have precedent for this in our code base, but am ok w/ it

Reply:
That is correct, I originally went with `detail::Program` but I realized that could make it unclear whether code referred to `tt_metal::v0::Program` or `tt_metal::detail::Program` in certain contexts, so I opted to adjust the name.